### PR TITLE
sycli: improved filter syntax

### PR DESCRIPTION
**Pull request by evanrichter**
this is my first try at #180, and everything seems to be working well!

The only thing I'm not sure of is how you want to do error handling on clearly invalid strings like "peers>five". This would currently panic with the message "Invalid numeric value"

### CODE REVIEW

File: sycli/src/main.rs [hunk 1]
```diff
@@ -665,3 +667,120 @@ fn main() {
...
+    // possible delimiters
+    let delimiters = vec![">=", "<=", "==", "!=", "<", ">", ":"];
+
+    'search: for search in searches.split_whitespace() {
+        for delim in &delimiters {
+            if search.split(delim).count() == 2 {
+                let field = search.split(delim).nth(0).unwrap().to_string();
+                let value = search.split(delim).nth(1).unwrap();
+
+                let value = if field == "name"
```
Comment (Luminarys) [inline sycli/src/main.rs]: I think it's a good idea to do a match here and be explicit about possible field names we could expect. Then we can print to stderr that an unrecognized field name was used if nothing matches.

File: sycli/src/main.rs [hunk 2]
```diff
@@ -665,3 +667,120 @@ fn main() {
...
+/// Parse search criteria out of a filter string
+fn parse_filter(searches: &str) -> Vec<Criterion> {
+    use rpc::criterion::{Operation, Value};
+
+    let mut criterion = Vec::new();
+
+    // possible delimiters
+    let delimiters = vec![">=", "<=", "==", "!=", "<", ">", ":"];
+
+    'search: for search in searches.split_whitespace() {
```
Comment (Luminarys) [inline sycli/src/main.rs]: I wonder if there's a convenient crate to do more advanced tokenization. receptor supports things like name:"spaced name", but it seems like this wouldn't work so well here. If it's very inconvenient to find a good looking library for this though I think this is good enough for now.

File: sycli/src/main.rs [hunk 3]
```diff
@@ -665,3 +667,120 @@ fn main() {
...
+                };
+
+                criterion.push(Criterion { field, op, value });
+                continue 'search;
+            }
+        }
+    }
+
+    // if no delimiters found, assume a simple name query
+    if criterion.is_empty() {
```
Comment (Luminarys) [inline sycli/src/main.rs:749]: Can you add a test which has a valid filter condition, then invalid, e.g. "name:foo key~val" to show this branch doesn't happen per loop iteration.

Comment (Luminarys) [reply sycli/src/main.rs:749]: Also can you add a test which shows that a filter like "spaced name" gets translated into this condition with its whitespace

**Review by Luminarys — COMMENTED** [review]
Thanks for doing this!

**Comment by evanrichter** [comment]
@Luminarys ok I think I addressed everything! At a high level, I'm using the regex crate now, building two regexes (one for string types and one for numeric), and now either can be quoted or unquoted, and just like before if nothing matches right, the string is put into a name:<input> search. Also I added a delimiter `::` for case-sensitive searches. `:` is still case insensitive

again, I'm glad to contribute and open to suggestions

**Comment by Luminarys** [comment]
Looks good now, thanks!
